test/BinaryParse.cpp

```diff
@@ -24,12 +24,421 @@
 // TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 // MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 
+#include "UnitSPIRV.h"
+
+#include "TestFixture.h"
+#include "gmock/gmock.h"
+
+// Returns true if two spv_parsed_operand_t values are equal.
+// To use this operator, this definition must appear in the same namespace
+// as spv_parsed_operand_t.
+static bool operator==(const spv_parsed_operand_t& a,
+                       const spv_parsed_operand_t& b) {
+  return a.offset == b.offset && a.num_words == b.num_words &&
+         a.type == b.type && a.number_kind == b.number_kind &&
+         a.number_bit_width == b.number_bit_width;
+}
+
 namespace {
 
+using ::spvtest::Concatenate;
+using ::spvtest::MakeInstruction;
+using ::spvtest::MakeVector;
+using ::testing::AnyOf;
+using ::testing::Eq;
+
+// An easily-constructible and comparable object for the contents of an
+// spv_parsed_instruction_t.
+struct ParsedInstruction {
+  ParsedInstruction(const spv_parsed_instruction_t& inst)
+      : words(inst.words, inst.words + inst.num_words),
+        opcode(inst.opcode),
+        ext_inst_type(inst.ext_inst_type),
+        type_id(inst.type_id),
+        result_id(inst.result_id),
+        operands(inst.operands, inst.operands + inst.num_operands) {}
+
+  std::vector<uint32_t> words;
+  SpvOp opcode;
+  spv_ext_inst_type_t ext_inst_type;
+  uint32_t type_id;
+  uint32_t result_id;
+  std::vector<spv_parsed_operand_t> operands;
+
+  bool operator==(const ParsedInstruction& b) const {
+    return words == b.words && opcode == b.opcode &&
+           ext_inst_type == b.ext_inst_type && type_id == b.type_id &&
+           result_id == b.result_id && operands == b.operands;
+  }
+};
+
+// Prints a ParsedInstruction object to the given output stream.
+void PrintTo(const ParsedInstruction& inst, std::ostream* os) {
```

Comment:
IIRC in order to get EXPECT* to print using the << operator you have to define
::std::ostream& operator<<(::std::ostream&, const ParsedInstruction&) rather than
ParsedInstruction::operator<<(::std::ostream&)
which is not allowed.